source/geometries/NextDemoTrackingPlane.cc

```diff
@@ -86,31 +86,52 @@ void NextDemoTrackingPlane::Construct()
   G4double mask_thickn     = 0.;
   G4double membrane_thickn = 0.;
   G4double coating_thickn  = 0.;
+  G4String hole_type       = "";
   G4double hole_diameter   = 0.;
+  G4double hole_x          = 0.;
+  G4double hole_y          = 0.;
+  G4String sipm_type       = "";
 
   if (config_ == "run5") {
     if(verbosity_) G4cout << "run5 ..." << G4endl;
     gate_board_dist = 16.76 * mm;
     mask_thickn     = 2.0 * mm;
     membrane_thickn = 0.0 * mm;
     coating_thickn  = 0.0 * mm;
+    hole_type       = "rounded";
     hole_diameter   = 3.5 * mm;
+    sipm_type       = "sensl";
   }
   else if (config_ == "run7") {
     if(verbosity_) G4cout << "run7 ..." << G4endl;
     gate_board_dist = 9.16 * mm;
     mask_thickn     = 2.0  * mm;
     membrane_thickn = 0.25 * mm;
     coating_thickn  = 2.0  * micrometer;
+    hole_type       = "rounded";
     hole_diameter   = 3.5  * mm;
+    sipm_type       = "sensl";
   }
   else if (config_ == "run8") {
     if(verbosity_) G4cout << "run8 ..." << G4endl;
     gate_board_dist = 19.66 * mm;
     mask_thickn     = 5.0  * mm;
     membrane_thickn = 0.2  * mm;
     coating_thickn  = 2.0  * micrometer;
+    hole_type       = "rounded";
     hole_diameter   = 4.0  * mm;
+    sipm_type       = "sensl";
+  }
+  else if (config_ == "run9") {
+    if(verbosity_) G4cout << "run9 ..." << G4endl;
+    gate_board_dist = 19.66 * mm;
+    mask_thickn     = 6.0 * mm;
+    membrane_thickn = 0.2 * mm;
+    coating_thickn  = 2.0 * micrometer;
+    hole_type       = "rectangular";
+    hole_x          = 6.0 * mm;
+    hole_y          = 5.0 * mm;
+    sipm_type       = "next100";
   }
```

Comment:
I would add a last `else` which raises an exception if `config_` takes none of the above values.

Reply:
In fact, I would move the exception at line 81 here.

Reply:
Looking at the rest of the code, I see that the check that the configuration is a legal one is done in `NextDemoInnerElements`. In that case, I would say that iy is superfluous to add any exception here, so I would remove that at line 81, too. The same should be done in `NextDemoFieldCage`. Do you agree?

Reply:
Agree. We are redundantly doing the same check in three different places now.